corehq/apps/export/det/schema_generator.py

```diff
@@ -64,7 +64,15 @@ def __init__(self, dd_property_types):
 
     @staticmethod
     def transform_path(input_path):
-        # either return hard-coded lookup or add prefix
+
+        # todo: this could be made smarter in the future for multi-parentage use cases
+        def _is_index_reference(path):
+            path_as_array = path.split('.')
+            return len(path_as_array) == 2 and path_as_array[0] == 'indices'
+
+        if _is_index_reference(input_path):
+            return PARENT_INDEX_PATH  # as could this
```

Comment:
`indices.snafu -> indices.parent.case_id` feels a bit too open-ended. Thinking of Hyrums law here, will this make it hard to make a backward compatible implementation of `# todo: this could be made smarter in the future...`?

Reply:
> Thinking of Hyrums law here, will this make it hard to make a backward compatible implementation of # todo: this could be made smarter in the future...?

After looking up Hyrum's law, I don't think this applies, for a few reasons:

1. 99% of the time this behavior will be "correct". To my knowledge (and correct me if I'm wrong, I'm not up to speed on the latest app builder developments), case multi-parentage or custom index names are only possible via advanced modules or a few other advanced feature flags.
2. If the behavior isn't correct, it will be obviously a "bug", with an easily justifiable fix.
3. From a workflow perspective, this isn't an API, it's a reference file that is typically used as a starting point and then customized. So changing the API later doesn't "break" anyone.

Does that make sense?

Reply:
Seems OK for now but also shouldn't be hard to make this follow the same pattern as the export schema generation: https://github.com/dimagi/commcare-hq/blob/9120bc74d3c43e0fbfe814f3978a73e27b441274/corehq/apps/export/models/new.py#L2146-L2165

Reply:
Yeah, I didn't even attempt it yesterday since it wasn't declared a problem for GA release by product and the information needed wasn't readily available without a refactor. But I'll take a closer look and see if it's easy enough to support and make a call.

Reply:
Hmm, I might be missing something, but I'm not actually sure this is possible/easy given how we store things.

We store something that looks like this in the export doc:

```
{
                    "deid_transform": null,
                    "doc_type": "CaseIndexExportColumn",
                    "help_text": "The ID of the associated activity case type",
                    "is_advanced": false,
                    "is_deleted": false,
                    "item": {
                        "datatype": null,
                        "doc_type": "CaseIndexItem",
                        "inferred": false,
                        "inferred_from": [],
                        "label": "parent.activity",
                        "last_occurrences": {
                            "1c92b29059c9f7b217e4751a73ad159c": 82
                        },
                        "path": [
                            {
                                "doc_type": "PathNode",
                                "is_repeat": false,
                                "name": "indices"
                            },
                            {
                                "doc_type": "PathNode",
                                "is_repeat": false,
                                "name": "activity"
                            }
                        ],
                        "tag": "case",
                        "transform": null
                    },
                    "label": "indices.activity",
                    "selected": true,
                    "tags": [
                        "case"
                    ]
                },
```

And it resolves to [this code](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/export/models/new.py#L2611-L2617) which relies on actually having the `doc` in place to find the relevant indices.

The index "key" doesn't appear to be stored anywhere except maybe the "label" field (`parent.activity` in the above example). Do you know if it's valid to assume that `parent` in there represents the index ID?

Reply:
Ah, I tracked down [the code that sets that field](https://github.com/dimagi/commcare-hq/blob/9120bc74d3c43e0fbfe814f3978a73e27b441274/corehq/apps/export/models/new.py#L2199-L2205). It does appear to be a valid - if very opaque - assumption. But not worse than other things happening around these fields.

Reply:
ok it wasn't too bad so just added support for this